### ISSUE TITLE
About selected 2D detector & extended training data problem

### ISSUE THREAD

**Issue by wenwen12321** [paTRICK-swk/P-STMO#16]
Hi author,

I would appreciate it if you could help me understand the following two questions.

**(1)【Why not use the same 2D detector on all datasets (e.g. h36m, 3dhp, in-the-wild...)】**
I wonder why you (or most methods) use "AlphaPose" as the in-the-wild 2D keypoint detector while using "CPN" as the 2D detector of Human3.6M. 
Again, in the MPI-INF-3DHP dataset, why do not use the same 2D detector in Human3.6M (e.g. CPN) to generate 2D pose but use GT 2D poses as inputs? 

**(2) 【Extend training data by merging  h36m and 3dhp】**
The experiment setting in your paper is training on h36m training set and then evaluating on h36m test set; 
training on 3dhp training set and then evaluating on 3dhp test set.

I wonder if it is reasonable to merge h36m and 3dhp training set for more diverse training poses and scenes. If so, could you point me to how to pre-processing data in this case?

Thank you so much.
Look for your reply.

**Comment by paTRICK-swk** [comment]
(1) For me, when testing on in-the-wild videos, AlphaPose has better open-source support and is simple to implement. And when testing on the datasets, the use of CPN/GT for Human3.6M and GT for 3DHP is for benchmarking purposes, which ensures that the experimental setup is consistent when comparing our method with others. It's okay to report CPN results for 3DHP, except I don't do it. In fact, you can use any 2D pose detector you want.

(2) The two datasets differ in their labeling of joint positions. Although the 3DHP dataset provides the Human3.6M joint format (17 points), there are still subtle differences (e.g., the position of the ‘head’ joint). You can draw the 2D joint locations of these two datasets to see the differences.

You can try to use both parts of the data directly for training with the assurance that the joints are in the same order. However, this puts a burden on the training process and I am not sure if you can have a good result.

**Comment by wenwen12321** [comment]
I've got it! Thanks for the very informative response!!
